### PR TITLE
Fix script vars being dumb

### PR DESCRIPTION
**Pull request by elkowar**
This PR attempts to fix issues with script-var scripts not being closed correctly.

### CODE REVIEW

**Comment by elkowar** [comment]
@buffet could you maybe look over this again? I'm really unsure if i understand all of this, and i'd want to make sure that it's now _done_ and no longer an issue ^^

**Comment by buffet** [comment]
Remind me tomorrow.

**Comment by elkowar** [comment]
@buffet there ya go, reminder here!

File: src/script_var_handler.rs [hunk 1]
```diff
@@ -145,23 +145,43 @@ impl Drop for ScriptVarHandler {
...
+        let pipe = filedescriptor::Pipe::new()?;
+
+        match unsafe { fork()? } {
+            ForkResult::Child => {
+                std::mem::drop(pipe.read);
+                dup2(pipe.write.as_raw_fd(), std::io::stdout().as_raw_fd())?;
+                setpgid(Pid::from_raw(0), Pid::from_raw(0))?;
+                execv(
+                    CString::new("/bin/sh")?.as_ref(),
+                    &[CString::new("sh")?, CString::new("-c")?, CString::new(command)?],
```
Comment (buffet) [inline src/script_var_handler.rs:167]: This should be /bin/sh.

File: src/script_var_handler.rs [hunk 2]
```diff
@@ -145,23 +145,43 @@ impl Drop for ScriptVarHandler {
...
+                    pid: child,
+                    out_fd: pipe.read,
+                })
+            }
+        }
+    }
+
+    pub fn kill(self) {
+        unsafe {
+            nix::libc::kill(self.pid.as_raw(), libc::SIGTERM);
```
Comment (buffet) [inline src/script_var_handler.rs:184]: nix::sys::signal::kill takes the Pid newtype and is prolly what you should use here.

**Review by buffet — CHANGES_REQUESTED** [review]

**Comment by elkowar** [comment]
Thanks a ton, @buffet!

**Review by buffet — APPROVED** [review]

**Pull request merged**